### PR TITLE
Update benchmark_app to pass precision and layout via command line

### PR DESCRIPTION
**Pull request by ArtemySkrebkov-intel**
### Summary
* Move functions for parsing layout and precision in common place to re-use it in benchmark_app and compile_tool
* Extract common functionality for tests into samples_utils library

### Related ticket
* https://jira.devtools.intel.com/browse/CVS-47420

### CODE REVIEW

**Review by vinograd47 — APPROVED** [review]
Looks OK to me

**Comment by ArtemySkrebkov-intel** [comment]
As agreed with @nkogteva @ilya-lavrenov I removed layout support from benchmark_app since the change contradicts with https://github.com/openvinotoolkit/openvino/pull/4002 and requires further discussion.

@Maxim-Doronin 
There are changes in myriad tools. Please review.

**Comment by ilya-lavrenov** [comment]
Do we really need such huge changes just to add precision to benchmnark_app? O_o

**Comment by nkogteva** [comment]
We are keeping both python and C++ versions are aligned. Please update python version as well.

**Comment by ArtemySkrebkov-intel** [comment]
@ilya-lavrenov  
I could just add parsing to benchmark_app but it would duplicate code from compile_tool so I decided to re-use parsing of arguments from compile_tool to avoid duplication and have benchmark_app and compile_tool aligned. 
It required to move the parsing to a common place.

**Review by Maxim-Doronin — APPROVED** [review]

**Comment by ArtemySkrebkov-intel** [comment]
@nkogteva @ilya-lavrenov 
It is a good point. Do you mind if I update the python sample while porting this PR into `master`?
We need only cpp sample available for upcoming OpenVINO for VPUX platforms.

**Comment by nkogteva** [comment]
@ArtemySkrebkov-intel I think it is ok to align python tool during porting to master.

**Review by nkogteva — APPROVED** [review]

File: inference-engine/cmake/compile_options.cmake [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+function(enable_warnings_as_errors TARGET_NAME)
```
Comment (ilya-lavrenov) [inline inference-engine/cmake/compile_options.cmake:5]: 1. seems to be unused
2. it's supposed to be used in samples code. BUT: this file is not a part of OpenVINO package, so we will have cmake error if `enable_warnings_as_errors` is called from samples

File: inference-engine/samples/CMakeLists.txt [hunk 2]
```diff
@@ -224,7 +228,8 @@ macro(ie_add_sample)
     if(IE_SAMPLE_INCLUDE_DIRECTORIES)
         target_include_directories(${IE_SAMPLE_NAME} PRIVATE ${IE_SAMPLE_INCLUDE_DIRECTORIES})
     endif()
-    target_include_directories(${IE_SAMPLE_NAME} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/../common")
+
+     target_include_directories(${IE_SAMPLE_NAME} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/../common")
```
Comment (ilya-lavrenov) [inline inference-engine/samples/CMakeLists.txt:232]: leave as is, please

File: inference-engine/samples/common/utils/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+source_group(TREE ${CMAKE_CURRENT_SOURCE_DIR} FILES ${SOURCES})
+
+add_library(${TARGET_NAME} STATIC ${SOURCES})
+set_target_properties(${TARGET_NAME} PROPERTIES FOLDER "src")
+
+# FIXME: need to fix warnings on android
+#include(compile_options)
+#enable_warnings_as_errors(${TARGET_NAME} WIN_STRICT)
+
+ie_developer_export_targets(${TARGET_NAME})
```
Comment (ilya-lavrenov) [inline inference-engine/samples/common/utils/CMakeLists.txt:13]: not available in OpenVINO release package.

File: inference-engine/samples/common/utils/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+target_include_directories(${TARGET_NAME}
+    PUBLIC
+        "${CMAKE_CURRENT_SOURCE_DIR}/include"
+    PRIVATE
+        "${IE_MAIN_SOURCE_DIR}/src/vpu/common/include"
+        "${IE_MAIN_SOURCE_DIR}/src/plugin_api"
```
Comment (ilya-lavrenov) [inline inference-engine/samples/common/utils/CMakeLists.txt:20]: the same for 2 dirs above

**Review by ilya-lavrenov — CHANGES_REQUESTED** [review]

File: inference-engine/samples/common/utils/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,27 @@
+set(TARGET_NAME "samples_utils")
```
Comment (ilya-lavrenov) [inline inference-engine/samples/common/utils/CMakeLists.txt:1]: copyright header

File: inference-engine/samples/common/utils/CMakeLists.txt (hunk 3 above)
Comment (ilya-lavrenov) [reply inference-engine/samples/common/utils/CMakeLists.txt:13]: move to `openvino_developer_export_targets(COMPONENT openvino_common TARGETS format_reader ..)`

File: inference-engine/cmake/compile_options.cmake (hunk 1 above)
Comment (ilya-lavrenov) [reply inference-engine/cmake/compile_options.cmake:5]: remove this file

File: inference-engine/samples/benchmark_app/CMakeLists.txt [hunk 6]
```diff
@@ -8,5 +8,6 @@ file (GLOB HDR ${CMAKE_CURRENT_SOURCE_DIR}/*.hpp)
 ie_add_sample(NAME benchmark_app
               SOURCES ${SRC}
               HEADERS ${HDR}
-              DEPENDENCIES format_reader
+              DEPENDENCIES format_reader samples_utils
```
Comment (ilya-lavrenov) [inline inference-engine/samples/benchmark_app/CMakeLists.txt:11]: let's rename `samples_utils` to `ie_samples_utils`

**Comment by ilya-lavrenov** [comment]
Let's merge  `vpu_utils` and `os` to new `utils`

File: inference-engine/cmake/compile_options.cmake (hunk 1 above)
Comment (ArtemySkrebkov-intel) [reply inference-engine/cmake/compile_options.cmake]: Removed.

File: inference-engine/samples/CMakeLists.txt (hunk 2 above)
Comment (ArtemySkrebkov-intel) [reply inference-engine/samples/CMakeLists.txt]: Removed.

File: inference-engine/samples/common/utils/CMakeLists.txt (hunk 3 above)
Comment (ArtemySkrebkov-intel) [reply inference-engine/samples/common/utils/CMakeLists.txt]: Added https://github.com/openvinotoolkit/openvino/pull/4167/files#diff-72589b94b3e5aeacbc26e04d16dfd6692aa622cd77b8f09e146b85bb1369775aR183

File: inference-engine/samples/common/utils/CMakeLists.txt (hunk 4 above)
Comment (ArtemySkrebkov-intel) [reply inference-engine/samples/common/utils/CMakeLists.txt]: No need to have them anymore.

File: inference-engine/samples/common/utils/CMakeLists.txt (hunk 5 above)
Comment (ArtemySkrebkov-intel) [reply inference-engine/samples/common/utils/CMakeLists.txt]: Added

File: inference-engine/samples/benchmark_app/CMakeLists.txt (hunk 6 above)
Comment (ArtemySkrebkov-intel) [reply inference-engine/samples/benchmark_app/CMakeLists.txt]: Renamed.

**Comment by ArtemySkrebkov-intel** [comment]
@ilya-lavrenov  
Please take a look one more time

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by azhogov** [comment]
CI passed: https://gitlab-icv.inn.intel.com/inference-engine/product-configs/merge_requests/1378

**Pull request merged**